### PR TITLE
[plugin.video.haystacktv] 1.0.0

### PR DESCRIPTION
**Pull request by Lunatixz**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.haystacktv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <summary lang="en_GB"></summary>
+        <description lang="en_GB">Haystack TV is your personal headline news channel. We learn about your interests in order to surface the news that matters to you! Watch it on Kodi today...</description>
+        <platform>all</platform>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
+        <source>https://github.com/Lunatixz/XBMC_Addons/tree/master/plugin.video.haystacktv</source>
+        <website>http://www.haystack.tv/</website>
+        <forum>https://forum.kodi.tv/showthread.php?tid=316485</forum>
+        <news></news>
+        <assets>
+            <icon>icon.png</icon>
```
Comment (enen92) [inline plugin.video.haystacktv/addon.xml]: As this is a new submission and you are targeting krypton as a base version you should probably respect the new addon structure: move both files to the resources folder. See: http://kodi.wiki/view/Add-on_structure

File: plugin.video.haystacktv/default.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        self.getCategories()
+
+        
+    def openURL(self, url):
+        try:
+            cacheResponce = self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+            if not cacheResponce:
+                request = urllib2.Request(url)
+                request.add_header('User-Agent','Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)')
+                responce = urllib2.urlopen(request, timeout = TIMEOUT)
```
Comment (enen92) [inline plugin.video.haystacktv/default.py]: you have a typo here (but still using the typo on both calls). Just fyi.

File: plugin.video.haystacktv/addon.xml (hunk 1 above)
Comment (Lunatixz) [inline plugin.video.haystacktv/addon.xml]: Wow surprised I missed that, hole lines missing. My bad

**Comment by Lunatixz** [comment]
@enen92 i'm confused by fanart.jpg location change? isn't icon.png and fanart.jpg always in the root folder? or is its new location in a image folder?

**Comment by enen92** [comment]
@Lunatixz not for krypton and onwards. It still work but as a way to achive backwards compatibility.
Since this is a new addon that will only support versions higher than Krypton, there is no reason for you to keep it in the root folder.

File: plugin.video.haystacktv/default.py (hunk 2 above)
Comment (Lunatixz) [inline plugin.video.haystacktv/default.py]: haha missed that, thx for the heads up.

**Pull request merged**